app/src/org/commcare/views/dialogs/CustomProgressDialog.java

```diff
@@ -156,11 +157,9 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         TextView messageView = (TextView)view.findViewById(R.id.progress_dialog_message);
         messageView.setText(message);
 
-        if (usingCancelButton) {
-            Button cancelButton = setupCancelButton(view);
-            if (wasCancelPressed) {
-                setCancellingText(titleView, messageView, cancelButton);
-            }
+        setupCancelButton(view);
+        if (wasCancelPressed) {
+            setCancellingText(titleView, messageView, cancelButton);
```

Comment:
We can't guarantee that the action was cancellable when this was pressed, we likely need to wait for a signal that says "cancel accepted" or something similar.